src/core/toga/window.py

```diff
@@ -175,6 +187,10 @@ def full_screen(self, is_full_screen):
         self._impl.set_full_screen(is_full_screen)
 
     def on_close(self):
+        # If we registered this window with the app, make sure we remove
+        # that registration.
+        if self.app and self._app_id:
+            self.app.del_window(self._app_id)
```

Comment:
We need to check how this will interact with `on_close` and `app.on_exit` handlers - can a window (and, in particular, the main window) rely on the existence of `self.app` on a window when it is being closed?

Reply:
We might be able to. When a window is created, self.app is set to None, and hence the if statement will be false, and that step will be skipped.

In today's version of the code, the registry is no longer a dictionary, it's a set. And I use the discard() method rather than the remove() method. So if a window is being closed, and it has previously been registered, but for some reason app.windows no longer has the window mentioned, then the discard() will silently do no action.